aten/src/ATen/native/cpu/Unfold2d.cpp

```diff
@@ -255,6 +354,68 @@ static void unfolded2d_copy(
       });
 }
 
+template <typename scalar_t>
+static void unfolded2d_copy_channels_last(
+    scalar_t* input_data,
+    scalar_t* finput_data,
+    int64_t kH,
+    int64_t kW,
+    int64_t dH,
+    int64_t dW,
+    int64_t padH,
+    int64_t padW,
+    int64_t batch_size,
+    int64_t n_input_plane,
+    int64_t input_height,
+    int64_t input_width,
+    int64_t output_height,
+    int64_t output_width) {
+  // algorithm on channels last is dual problem to contiguous format...
```

Comment:
Just curious on this comment. On what aspects, you define this algorithm as a dual problem to the contiguous format?

Reply:
The output shape of Unfold (or im2col) in ChannelsLast format is transpose of Contiguous format.

Suppose `batch_size=1` to simply:
```bash
### contiguous (nchw), tensor shape in 2d view:
input: [input_channel, input_height * input width]
column: [input_channel * kernel_height * kernel_width, output_height * output_width]
weight: [output_channel, input_channel * kernel_height * kernel_width]
output = weight * column

### channels last (nhwc), tensor shape in 2d view:
input: [input_height * input_width, input_channel]
column: [output_height * output_width, kernel_height * kernel_width * input_channel]
weight: [output_channel, kernel_height * kernel_width * input_channel]
output = column * weight.transpose()
```